awx/main/utils/encryption.py

```diff
@@ -98,7 +104,13 @@ def decrypt_field(instance, field_name, subfield=None):
     '''
     Return content of the given instance and field name decrypted.
     '''
-    value = getattr(instance, field_name)
+    try:
+        value = instance.inputs[field_name]
+    except (TypeError, AttributeError):
+        value = getattr(instance, field_name)
```

Comment:
These utilities always raised an `AttributeError` when the provided field_name was undefined on the model, but I've added some unit tests to make sure that stays intact.

The `try` clause is covered by our functional test in `test_credentials`, as is the handler for `KeyError` - the test couldn't pass if either weren't working as expected.